tests/decodeoutput.cpp

```diff
@@ -738,6 +741,205 @@ DecodeOutputDmabuf::DecodeOutputDmabuf(VideoDataMemoryType memoryType)
 #endif //__ENABLE_TESTS_GLES__
 #endif //__ENABLE_X11__
 
+#ifdef __ENABLE_WAYLAND__
+struct display {
+    struct wl_display       *display;
+    struct wl_registry      *registry;
+    struct wl_compositor    *compositor;
+    struct wl_shell         *shell;
+    struct wl_shell_surface *shell_surface;
+    struct wl_surface       *surface;
```

Comment:
Could we use` SharedPtr<wl_surface> surface` for this? So we do not need care about release.
Just like this
`d->surface.reset(wl_compositor_create_surface(d->compositor), wl_surface_destroy);`
Same thing applied to all fields above.